cmake/AutodetectLibrary.cmake

```diff
@@ -0,0 +1,164 @@
+#
+# Try to autodetect a library via various of means.
+#
+# We first assume that find_package() can find it.
+# Next we try it via pkg-config.
+# If all that fails, do a find_path()/find_library(), as last effort.
+#
+# This is done as we support a wide variaty of systems, and not all have the
+# CMake files we need. For example, lzo doesn't even have pkg-config files
+# in Debian Stretch (but does in Debian Sid).
+#
+
+macro(_autodetect_library_via_pkgconfig NAME PKGCONFIG)
+    include(FindPkgConfig QUIET)
+
+    if (PKG_CONFIG_FOUND)
+        pkg_search_module("${NAME}" "${PKGCONFIG}" QUIET)
+
+        # pkg-config returns the libraries without really resolving them.
+        # This has as huge drawback that sometimes you only notice on linking
+        # a library is missing. To do better detecting during configuration,
+        # run all the libraries pkg-config returns through detection again, to
+        # see if they really exist.
+        # This is also how most CMake files work, and fixes issues on OSX where
+        # libraries are otherwise not always detected (because CMake usees more
+        # library paths than the compiler normally does).
+        if (${NAME}_FOUND)
+            unset(PKGCONFIG_LIBRARIES)
+            foreach(PKGCONFIG_LIBRARY ${${NAME}_LIBRARIES})
+                unset(FIND_PKGCONFIG_LIBRARY CACHE)
+                find_library(FIND_PKGCONFIG_LIBRARY NAMES "${PKGCONFIG_LIBRARY}")
+
+                if (FIND_PKGCONFIG_LIBRARY STREQUAL "FIND_PKGCONFIG_LIBRARY-NOTFOUND")
+                    message(FATAL_ERROR "pkg-config indicates '${PKGCONFIG}' depends on library '${PKGCONFIG_LIBRARY}' but this library was not found on the system")
+                endif (FIND_PKGCONFIG_LIBRARY STREQUAL "FIND_PKGCONFIG_LIBRARY-NOTFOUND")
+
+                list(APPEND PKGCONFIG_LIBRARIES ${FIND_PKGCONFIG_LIBRARY})
+            endforeach()
+            unset(FIND_PKGCONFIG_LIBRARY CACHE)
+
+            set(${NAME}_LIBRARIES "${PKGCONFIG_LIBRARIES}")
+        endif (${NAME}_FOUND)
+    endif (PKG_CONFIG_FOUND)
+endmacro()
+
+macro(_autodetect_library_via_find NAME HEADER_PATH HEADER LIBRARY)
+    find_path(${NAME}_INCLUDE_DIR NAMES "${HEADER}" PATH_SUFFIXES "${HEADER_PATH}")
+    find_library(${NAME}_LIBRARY NAMES "${LIBRARY}")
+
+    if (${NAME}_INCLUDE_DIR AND ${NAME}_LIBRARY)
+        set(${NAME}_FOUND YES)
+        set(${NAME}_INCLUDE_DIRS "${${NAME}_INCLUDE_DIR}")
+        set(${NAME}_INCLUDE_LIBRARIES "${${NAME}_LIBRARY}")
+    endif (${NAME}_INCLUDE_DIR AND ${NAME}_LIBRARY)
+endmacro()
+
+macro(_patch_older_cmake NAME)
+    # Patch up libraries that still announce under the deprecated name only
```

Comment:
Just require a newer version of CMake. If you still find modules that behave like this file a bug.